app/com/arpnetworking/metrics/portal/alerts/impl/DatabaseAlertRepository.java

```diff
@@ -0,0 +1,356 @@
+import javax.persistence.PersistenceException;
+
+/**
+ * Implementation of <code>AlertRepository</code> using SQL database.
+ *
+ * @author Deepika Misra (deepika at groupon dot com)
+ */
+public class DatabaseAlertRepository implements AlertRepository {
+
+    /**
+     * Public constructor.
+     *
+     * @param environment Play's <code>Environment</code> instance.
+     * @param config Play's <code>Configuration</code> instance.
+     * @throws Exception If the configuration is invalid.
+     */
+    @Inject
+    public DatabaseAlertRepository(final Environment environment, final Configuration config) throws Exception {
+        this(
+                ConfigurationHelper.<AlertQueryGenerator>getType(
+                        environment,
+                        config,
+                        "alertRepository.alertQueryGenerator.type")
+                        .newInstance());
+    }
+
+    /**
+     * Public constructor.
+     *
+     * @param alertQueryGenerator Instance of <code>AlertQueryGenerator</code>.
+     */
+    public DatabaseAlertRepository(final AlertQueryGenerator alertQueryGenerator) {
+        _alertQueryGenerator = alertQueryGenerator;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void open() {
+        assertIsOpen(false);
+        LOGGER.debug().setMessage("Opening alert repository").log();
+        _isOpen.set(true);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void close() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Closing alert repository").log();
+        _isOpen.set(false);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Optional<Alert> get(final UUID identifier) {
+        assertIsOpen();
+        LOGGER.debug()
+                .setMessage("Getting alert")
+                .addData("alertId", identifier)
+                .log();
+
+        final models.ebean.Alert ebeanAlert = Ebean.find(models.ebean.Alert.class)
+                .where()
+                .eq("uuid", identifier)
+                .findUnique();
+        if (ebeanAlert == null) {
+            return Optional.empty();
+        }
+        return Optional.of(convertFromEbeanAlert(ebeanAlert));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AlertQuery createQuery() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Preparing query").log();
+        return new DefaultAlertQuery(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public QueryResult<Alert> query(final AlertQuery query) {
+        assertIsOpen();
+        LOGGER.debug()
+                .setMessage("Querying")
+                .addData("query", query)
+                .log();
+
+        // Create the base query
+        final PagedList<models.ebean.Alert> pagedAlerts = _alertQueryGenerator.createAlertQuery(query);
+
+        // Compute the etag
+        // TODO(deepika)[ISSUE-7]: Obfuscate the etag
```

Comment:
Format is:

TODO(<USER>): Obfuscate the etag [ISSUE-<#>]

e.g.
TODO(deepika): Obfuscate the etag [ISSUE-7]